File: week-2/jewels-and-stones/jewels-and-stones.js
/**
 * @param {string} J
 * @param {string} S
 * @return {number}
 */
var numJewelsInStones = function(J, S) {
    const hashtable = {};
    for (let i of J) {
        if (i in hashtable)
            hashtable[i]++;
        else
            hashtable[i] = 1;
    }
    let res = 0;
    for (let i of S) {
        if (i in hashtable) res++;
    }
    return res;
};

File: week-2/jewels-and-stones/jewels-and-stones.py
class Solution:
    def numJewelsInStones(self, J: str, S: str) -> int:
        hashtable = {}
        for i in J:
            hashtable[i] = hashtable[i] + 1 if i in hashtable else 1
        
        res = 0
        for i in S:
            if i in hashtable:
                res += 1
        
        return res
